src/cunumeric/unary/scalar_unary_red_template.inl

```diff
@@ -116,15 +121,123 @@ struct ScalarUnaryRed {
+  {
+    rect   = args.in.shape<DIM>();
+    origin = rect.lo;
+    in     = args.in.read_accessor<RHS, DIM>(rect);
+    volume = pitches.flatten(rect);
+    shape  = args.shape;
+
+    out = args.out.reduce_accessor<LG_OP, true, 1>();
+    if constexpr (OP_CODE == UnaryRedCode::CONTAINS) { to_find = args.args[0].scalar<RHS>(); }
+
+    if ((args.has_where == true) && (OP_CODE != UnaryRedCode::CONTAINS)) {
+      where = args.where.read_accessor<bool, DIM>(rect);
+#ifndef LEGATE_BOUNDS_CHECKS
+      if (in.accessor.is_dense_row_major(rect) && where.accessor.is_dense_row_major(rect)) {
+        dense    = true;
+        inptr    = in.ptr(rect);
+        whereptr = where.ptr(rect);
+      }
+#endif
+    } else {
+#ifndef LEGATE_BOUNDS_CHECKS
+      // Check to see if this is dense or not
+      if (in.accessor.is_dense_row_major(rect)) {
+        dense = true;
+        inptr = in.ptr(rect);
+      }
+#endif
+    }
+  }
+
+  __CUDA_HD__ void operator()(LHS& lhs, size_t idx, LHS identity, DenseReduction) const noexcept
+  {
+    if constexpr (OP_CODE == UnaryRedCode::CONTAINS) {
+      if (inptr[idx] == to_find) { lhs = true; }
+    } else if constexpr (OP_CODE == UnaryRedCode::ARGMAX || OP_CODE == UnaryRedCode::ARGMIN ||
+                         OP_CODE == UnaryRedCode::NANARGMAX || OP_CODE == UnaryRedCode::NANARGMIN) {
+      auto p = pitches.unflatten(idx, origin);
+      if (whereptr[idx] == true)
+        OP::template fold<true>(lhs, OP::convert(p, shape, identity, inptr[idx]));
+    } else {
+      if (whereptr[idx] == true) OP::template fold<true>(lhs, OP::convert(inptr[idx], identity));
+    }
+  }
+
+  __CUDA_HD__ void operator()(LHS& lhs, size_t idx, LHS identity, SparseReduction) const noexcept
+  {
+    if constexpr (OP_CODE == UnaryRedCode::CONTAINS) {
+      auto point = pitches.unflatten(idx, origin);
+      if (in[point] == to_find) { lhs = true; }
+    } else if constexpr (OP_CODE == UnaryRedCode::ARGMAX || OP_CODE == UnaryRedCode::ARGMIN ||
+                         OP_CODE == UnaryRedCode::NANARGMAX || OP_CODE == UnaryRedCode::NANARGMIN) {
+      auto p = pitches.unflatten(idx, origin);
+      if (where[p] == true) OP::template fold<true>(lhs, OP::convert(p, shape, identity, in[p]));
+    } else {
+      auto p = pitches.unflatten(idx, origin);
+      if (where[p] == true) OP::template fold<true>(lhs, OP::convert(in[p], identity));
+    }
+  }
+
+  void execute() const noexcept
+  {
+    auto identity = LG_OP::identity;
+#ifndef LEGATE_BOUNDS_CHECKS
+    // The constexpr if here prevents the DenseReduction from being instantiated for GPU kernels
+    // which limits compile times and binary sizes.
+    if constexpr (KIND != VariantKind::GPU) {
+      // Check to see if this is dense or not
+      if (dense) {
+        return ScalarReductionPolicy<KIND, LG_OP, DenseReduction>()(volume, out, identity, *this);
+      }
+    }
+#endif
+    return ScalarReductionPolicy<KIND, LG_OP, SparseReduction>()(volume, out, identity, *this);
```

Comment:
Hmm, returning something (even if that something is `void`), is a bit of a code smell.
```suggestion
       ScalarReductionPolicy<KIND, LG_OP, DenseReduction>()(volume, out, identity, *this);
       return;
      }
    }
#endif
   ScalarReductionPolicy<KIND, LG_OP, SparseReduction>()(volume, out, identity, *this);
```